### PR TITLE
Ignore changes to target group arns in consul-cluster ASG

### PR DESCRIPTION
**Pull request by andrew-womeldorf**
Related to https://github.com/hashicorp/terraform-aws-vault/issues/210

A similar PR into `terraform-aws-vault` [is here](https://github.com/hashicorp/terraform-aws-vault/pull/211)

### CODE REVIEW

File: modules/consul-cluster/main.tf [hunk 1]
```diff
@@ -47,6 +47,10 @@ resource "aws_autoscaling_group" "autoscaling_group" {
       var.tags,
     ]
   )
+
+  lifecycle {
+    ignore_changes = [load_balancers, target_group_arns]
```
Comment (brikis98) [inline modules/consul-cluster/main.tf:52]: Could you add a comment explaining why this is here, including link(s) to the relevant docs / bugs?

**Review by brikis98 — COMMENTED** [review]
Thanks for the PR!

File: modules/consul-cluster/main.tf (hunk 1 above)
Comment (brikis98) [reply modules/consul-cluster/main.tf:60]: Perfect, thanks! I'll kick off tests now.

**Comment by brikis98** [comment]
All the tests failed with this sort of error:

```
TestConsulClusterWithUbuntu18Ami 2020-08-31T08:57:07Z command.go:168: [1m[31mError: [0m[0m[1mConflictsWith[0m
TestConsulClusterWithUbuntu18Ami 2020-08-31T08:57:07Z command.go:168: 
TestConsulClusterWithUbuntu18Ami 2020-08-31T08:57:07Z command.go:168: [0m  on modules/consul-cluster/main.tf line 19, in resource "aws_autoscaling_group" "autoscaling_group":
TestConsulClusterWithUbuntu18Ami 2020-08-31T08:57:07Z command.go:168:   19:   availability_zones  = [4mvar.availability_zones[0m
TestConsulClusterWithUbuntu18Ami 2020-08-31T08:57:07Z command.go:168: [0m
TestConsulClusterWithUbuntu18Ami 2020-08-31T08:57:07Z command.go:168: "availability_zones": conflicts with vpc_zone_identifier
TestConsulClusterWithUbuntu18Ami 2020-08-31T08:57:07Z command.go:168: [0m[0m
```

I think this was fixed in https://github.com/hashicorp/terraform-aws-consul/pull/182... Were you perhaps on an older version of the code? If so, could you rebase on the latest from master?

**Comment by andrew-womeldorf** [comment]
Yes, it's definitely possible I was on an older version of the code.
I pulled master and rebased my branch. Would you mind kicking off the tests again?

**Comment by brikis98** [comment]
Re-running tests now!

**Comment by brikis98** [comment]
Hm, getting some test failures. The cluster deploys, but the test doesn't seem to be able to connect:

```
TestConsulClusterWithEncryptionUbuntu18Ami 2020-09-01T11:09:59Z retry.go:72: Check Consul members
TestConsulClusterWithEncryptionUbuntu18Ami 2020-09-01T11:09:59Z retry.go:84: Check Consul members returned an error: Get http://54.175.203.50:8500/v1/agent/members: dial tcp 54.175.203.50:8500: connect: connection refused. Sleeping for 10s and will try again.
TestConsulClusterWithEncryptionUbuntu18Ami 2020-09-01T11:10:09Z retry.go:72: Check Consul members
TestConsulClusterWithEncryptionUbuntu18Ami 2020-09-01T11:10:09Z retry.go:84: Check Consul members returned an error: Get http://54.175.203.50:8500/v1/agent/members: dial tcp 54.175.203.50:8500: connect: connection refused. Sleeping for 10s and will try again.
TestConsulClusterWithEncryptionUbuntu18Ami 2020-09-01T11:10:19Z retry.go:72: Check Consul members
TestConsulClusterWithEncryptionUbuntu18Ami 2020-09-01T11:10:19Z retry.go:84: Check Consul members returned an error: Get http://54.175.203.50:8500/v1/agent/members: dial tcp 54.175.203.50:8500: connect: connection refused. Sleeping for 10s and will try again.
TestConsulClusterWithEncryptionUbuntu18Ami 2020-09-01T11:10:29Z retry.go:72: Check Consul members
TestConsulClusterWithEncryptionUbuntu18Ami 2020-09-01T11:10:29Z retry.go:84: Check Consul members returned an error: Get http://54.175.203.50:8500/v1/agent/members: dial tcp 54.175.203.50:8500: connect: connection refused. Sleeping for 10s and will try again.
TestConsulClusterWithEncryptionUbuntu18Ami 2020-09-01T11:10:39Z retry.go:72: Check Consul members
TestConsulClusterWithEncryptionUbuntu18Ami 2020-09-01T11:10:39Z retry.go:84: Check Consul members returned an error: Get http://54.175.203.50:8500/v1/agent/members: dial tcp 54.175.203.50:8500: connect: connection refused. Sleeping for 10s and will try again.
TestConsulClusterWithEncryptionUbuntu18Ami 2020-09-01T11:10:49Z retry.go:72: Check Consul members
TestConsulClusterWithEncryptionUbuntu18Ami 2020-09-01T11:10:49Z retry.go:84: Check Consul members returned an error: Get http://54.175.203.50:8500/v1/agent/members: dial tcp 54.175.203.50:8500: connect: connection refused. Sleeping for 10s and will try again.
```

Perhaps something is failing to boot up on the server? Were you able to get everything working in local testing? Did the server nodes come up and successfully form a quorum?

I also just re-ran the tests to see if it could've been a transient / timing issue.

**Comment by brikis98** [comment]
OK, tests passed on re-run. Must have been some transient issue. Merging now!

**Pull request merged**